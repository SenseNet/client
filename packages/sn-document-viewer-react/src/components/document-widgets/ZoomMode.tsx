import Divider from '@material-ui/core/Divider'
import IconButton from '@material-ui/core/IconButton'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import MobileStepper from '@material-ui/core/MobileStepper'

import AspectRatio from '@material-ui/icons/AspectRatio'
import Code from '@material-ui/icons/Code'
import Error from '@material-ui/icons/Error'

import ZoomIn from '@material-ui/icons/ZoomIn'
import ZoomOut from '@material-ui/icons/ZoomOut'
import ZoomOutMap from '@material-ui/icons/ZoomOutMap'

import React, { useCallback, useRef, useState } from 'react'
import { useLocalization, useViewerState } from '../../hooks'
import { ZoomMode } from '../../models/viewer-state'

/**
 * Document widget component for modifying the zoom mode / level
 */
export const ZoomModeWidget: React.FC = () => {
  const localization = useLocalization()
  const viewerState = useViewerState()

  const zoomMenuAnchor = useRef()
  const [isZoomMenuOpened, setIsZoomMenuOpened] = useState(false)

  const openZoomMenu = useCallback(() => {
    setIsZoomMenuOpened(true)
  }, [])

  const closeZoomMenu = (newZoomMode?: ZoomMode) => {
    if (newZoomMode) {
      viewerState.updateState({ zoomMode: newZoomMode })
    }
    setIsZoomMenuOpened(false)
  }

  const zoomIn = useCallback(
    (ev: React.MouseEvent<HTMLElement>) => {
      ev.preventDefault()
      ev.stopPropagation()
      viewerState.updateState({ customZoomLevel: viewerState.customZoomLevel + 1 || 1 })
    },
    [viewerState],
  )

  const zoomOut = useCallback(
    (ev: React.MouseEvent<HTMLElement>) => {
      ev.preventDefault()
      ev.stopPropagation()
      viewerState.updateState({ customZoomLevel: viewerState.customZoomLevel + 1 || 0 })
    },
    [viewerState],
  )

  return (
    <div style={{ display: 'inline-block' }}>
      <IconButton onClick={openZoomMenu} title={localization.zoomMode} innerRef={zoomMenuAnchor}>
        {(() => {
          switch (viewerState.zoomMode) {
            case 'custom':
              if (viewerState.customZoomLevel > 0) {
                return <ZoomIn />
              }
              return <ZoomOut />
            case 'fitHeight':
              return <Code style={{ transform: 'rotate(90deg)' }} />
            case 'fitWidth':
              return <Code />
            case 'originalSize':
              return <AspectRatio />
            case 'fit':
              return <ZoomOutMap />
            default:
              return <Error />
          }
        })()}
      </IconButton>
      <Menu id="zoom-menu" anchorEl={zoomMenuAnchor.current} open={isZoomMenuOpened} onClose={() => closeZoomMenu()}>
        <MenuItem onClick={() => closeZoomMenu('fit')}>
          <ZoomOutMap /> &nbsp; {localization.zoomModeFit}{' '}
        </MenuItem>
        <MenuItem onClick={() => closeZoomMenu('originalSize')}>
          <AspectRatio />
          &nbsp; {localization.zoomModeOriginalSize}{' '}
        </MenuItem>
        <MenuItem onClick={() => closeZoomMenu('fitHeight')}>
          <Code style={{ transform: 'rotate(90deg)' }} /> &nbsp; {localization.zoomModeFitHeight}{' '}
        </MenuItem>
        <MenuItem onClick={() => closeZoomMenu('fitWidth')}>
          <Code /> &nbsp; {localization.zoomModeFitWidth}{' '}
        </MenuItem>
        <Divider light={true} />
        &nbsp; {localization.zooomModeCustom} <br />
        <MobileStepper
          variant="progress"
          steps={6}
          position="static"
          activeStep={viewerState.customZoomLevel}
          nextButton={
            <IconButton disabled={viewerState.customZoomLevel === 5} onClickCapture={zoomIn}>
              <ZoomIn />
            </IconButton>
          }
          backButton={
            <IconButton disabled={viewerState.customZoomLevel === 0} onClickCapture={zoomOut}>
              <ZoomOut />
            </IconButton>
          }
        />
      </Menu>
    </div>
  )
}
