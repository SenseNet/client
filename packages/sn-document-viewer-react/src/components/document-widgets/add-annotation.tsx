import { createStyles, makeStyles, Theme } from '@material-ui/core'
import { ChatSharp } from '@material-ui/icons'
import clsx from 'clsx'
import React from 'react'
import { useDocumentPermissions, useLocalization, useViewerState } from '../../hooks'
import { ToggleBase } from './toggle-base'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    iconButton: {},
    icon: {},
    iconActive: {
      fill: theme.palette.primary.main,
    },
  })
})

type AddAnnnotationClassKey = Partial<ReturnType<typeof useStyles>>

/**
 * Document widget component to draw Annotation
 */
export const AddAnnotationWidget: React.FC<{ classes?: AddAnnnotationClassKey }> = (props) => {
  const classes = useStyles(props)
  const localization = useLocalization()
  const viewerState = useViewerState()
  const permissions = useDocumentPermissions()

  const pageRotation =
    viewerState.rotation?.find((rotation) => rotation.pageNum === viewerState.activePage)?.degree || 0

  return (
    <ToggleBase
      disabled={!permissions.canEdit || pageRotation !== 0}
      classes={classes}
      active={viewerState.activeShapePlacing === 'annotation'}
      title={localization.addAnnotation}
      setValue={(value) => viewerState.updateState({ activeShapePlacing: value ? 'annotation' : 'none' })}>
      <ChatSharp
        className={clsx(classes.icon, { [classes.iconActive]: viewerState.activeShapePlacing === 'annotation' })}
      />
    </ToggleBase>
  )
}
