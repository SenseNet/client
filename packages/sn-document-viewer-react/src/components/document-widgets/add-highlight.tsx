import { createStyles, makeStyles, Theme } from '@material-ui/core'
import ChatBubbleOutlineSharp from '@material-ui/icons/ChatBubbleOutlineSharp'
import clsx from 'clsx'
import React from 'react'
import { useDocumentPermissions, useLocalization, useViewerState } from '../../hooks'
import { ToggleBase } from './toggle-base'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    iconButton: {},
    icon: {},
    iconActive: {
      fill: theme.palette.primary.main,
    },
  })
})

type AddRHightlightClassKey = Partial<ReturnType<typeof useStyles>>

/**
 * Document widget component to draw Hihgligh
 */
export const AddHighlightWidget: React.FC<{ classes?: AddRHightlightClassKey }> = (props) => {
  const classes = useStyles(props)
  const localization = useLocalization()
  const viewerState = useViewerState()
  const permissions = useDocumentPermissions()

  return (
    <ToggleBase
      disabled={!permissions.canEdit}
      classes={classes}
      isVisible={viewerState.isPlacingHighlight}
      title={localization.addHighlight}
      setValue={(value) => viewerState.updateState({ isPlacingHighlight: value })}>
      <ChatBubbleOutlineSharp
        className={clsx(classes.icon, { [classes.iconActive]: viewerState.isPlacingHighlight })}
      />
    </ToggleBase>
  )
}
