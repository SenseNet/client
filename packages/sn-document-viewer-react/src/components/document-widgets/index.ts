export * from './add-annotation'
export * from './add-highlight'
export * from './add-redaction'
export * from './document-title-pager'
export * from './download-widget'
export * from './print-widget'
export * from './rotate-active-pages'
export * from './rotate-document'
export * from './rotate-widget'
export * from './save-widget'
export * from './share-widget'
export * from './toggle-base'
export * from './toggle-comments'
export * from './toggle-redaction'
export * from './toggle-shapes'
export * from './toggle-thumbnails'
export * from './toggle-watermark'
export * from './zoom-in-out'
