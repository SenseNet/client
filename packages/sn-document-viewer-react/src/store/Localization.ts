import { Reducer } from 'redux'
import { PreviewState } from '../Enums'

/**
 * Model definition for the localization store
 */
export interface LocalizationStateType {
  rotateDocumentLeft: string
  rotateDocumentRight: string
  rotatePageLeft: string
  rotatePageRight: string
  zoomMode: string
  zoomModeFit: string
  zoomModeOriginalSize: string
  zoomModeFitHeight: string
  zoomModeFitWidth: string
  zooomModeCustom: string
  toggleRedaction: string
  toggleWatermark: string
  toggleShapes: string
  toggleThumbnails: string
  toggleComments: string
  firstPage: string
  previousPage: string
  gotoPage: string
  nextPage: string
  lastPage: string
  saveChanges: string
  loadingDocument: string
  errorLoadingDocument: Array<{ state: PreviewState; message: string; details: string; code?: number }>
  errorLoadingDetails: string
  reloadPage: string
  search: string
  share: string
  download: string
  print: string
  showMore: string
  showLess: string
  avatarAlt: string
  delete: string
  deleteCommentDialogBody: string
  deleteCommentDialogTitle: string
  okButton: string
  cancelButton: string
  addComment: string
  commentInputPlaceholder: string
  submit: string
  inputRequiredError: string
  markerRequiredError: string
  markerTooltip: string
  commentSideBarTitle: string
}

/**
 * Default localization string value
 */
export const defaultLocalization: LocalizationStateType = {
  rotateDocumentLeft: 'Rotate document left',
  rotateDocumentRight: 'Rotate document right',
  rotatePageLeft: 'Rotate page left',
  rotatePageRight: 'Rotate page right',
  zoomMode: 'Zoom mode',
  zoomModeFit: 'Fit',
  zoomModeOriginalSize: 'Original size',
  zoomModeFitHeight: 'Fit height',
  zoomModeFitWidth: 'Fit width',
  zooomModeCustom: 'Custom',
  toggleRedaction: 'Toggle redaction',
  toggleWatermark: 'Toggle watermark',
  toggleShapes: 'Toggle shapes',
  toggleThumbnails: 'Toggle thumbnails',
  toggleComments: 'Toggle comments',
  firstPage: 'First page',
  previousPage: 'Previous page',
  gotoPage: 'Goto page',
  nextPage: 'Next page',
  lastPage: 'Last page',
  saveChanges: 'Save changes',
  loadingDocument: 'Preview image generation is in progress',
  errorLoadingDocument: [
    { state: PreviewState.Empty, message: "No preview available, because the document is empty!", details: '' },
    {
      code: 500,
      state: PreviewState.Empty,
      message: 'Ooops! Something went wrong...',
      details:
        'An unexpected error seems to have occured. Why not try refreshing your page? If the page still not working try to check back later.',
    },
    {
      code: 404,
      state: PreviewState.Empty,
      message: 'Ooops! Something went wrong...',
      details: `You don't have permission to see this document \r\n The document you are trying to open has been deleted \r\n The document has been moved to another place `,
    },
    { state: PreviewState.UploadFailure, message: 'Error during preview generation! No preview available.', details: '' },
    { state: PreviewState.UploadFailure2, message: 'Error during preview generation! No preview available.', details: '' },
    {
      state: PreviewState.ExtensionFailure,
      message: 'No preview available, because this type of file is not supported!',
      details: '',
    },
    { state: PreviewState.NoPreviewProviderEnabled, message: 'There is no preview provider enabled', details: '' },
  ],
  errorLoadingDetails: 'The following error occured during opening a document: ',
  reloadPage: 'Reload page',
  search: 'Search',
  share: 'Share',
  download: 'Download',
  print: 'Print',
  showMore: '+ Show more',
  showLess: '+ Show less',
  avatarAlt: 'Picture of the commenter',
  delete: 'Delete',
  deleteCommentDialogTitle: 'Delete comment?',
  deleteCommentDialogBody: `1 comment is about to permanently deleted. <br /><strong>Warning: You can't undo this action.</strong>`,
  okButton: 'Ok',
  cancelButton: 'Cancel',
  addComment: '+ Add Comment',
  commentInputPlaceholder: 'Make a comment',
  submit: 'Submit',
  inputRequiredError: 'The comment text is a required field.',
  markerRequiredError: 'You must place the marker first.',
  markerTooltip: 'You can put a marker with this button',
  commentSideBarTitle: 'Comments',
}

/**
 * Action that updates the store with new localized values
 * @param localization The new localization values
 */
export const setLocalization = (localization: Partial<LocalizationStateType>) => ({
  type: 'SN_DOCVIEWER_SET_LOCALIZATION',
  localization,
})

/**
 * Reducer for localization
 * @param state the current state
 * @param action the action to dispatch
 */
export const localizationReducer: Reducer<LocalizationStateType> = (state = defaultLocalization, action) => {
  switch (action.type) {
    case 'SN_DOCVIEWER_SET_LOCALIZATION': {
      return {
        ...state,
        ...action.localization,
      }
    }
    default:
      return state
  }
}
