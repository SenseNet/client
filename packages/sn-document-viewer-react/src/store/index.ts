import { Action, applyMiddleware, compose, createStore, Store } from 'redux'
import { ReduxDiMiddleware } from 'redux-di-middleware'
import { DocumentViewerSettings } from '../models'
import { rootReducer, RootReducerType } from './RootReducer'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

/**
 * gets a configuration object for the Store instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const getStoreConfig = (settings: DocumentViewerSettings) => {
  const di = new ReduxDiMiddleware(settings.injector)
  di.setInjectable(settings)
  return {
    rootReducer,
    preloadedState: { sensenetDocumentViewer: { documentState: { isLoading: true } } },
    enhancer: composeEnhancers(applyMiddleware(di.getMiddleware())),
  }
}

/**
 * returns a Store object for a Document Viewer instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const configureStore: (settings: DocumentViewerSettings) => Store<RootReducerType> = settings => {
  const config = getStoreConfig(settings)
  return createStore<RootReducerType, Action, {}, {}>(config.rootReducer, config.preloadedState, config.enhancer)
}

export * from './Comments'
export * from './Document'
export * from './PreviewImages'
export * from './RootReducer'
export * from './Viewer'
