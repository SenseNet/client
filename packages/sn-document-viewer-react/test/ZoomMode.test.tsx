import IconButton from '@material-ui/core/IconButton'
import Menu from '@material-ui/core/Menu'
import AspectRatio from '@material-ui/icons/AspectRatio'
import Code from '@material-ui/icons/Code'
import ZoomIn from '@material-ui/icons/ZoomIn'
import ZoomOut from '@material-ui/icons/ZoomOut'
import ZoomOutMap from '@material-ui/icons/ZoomOutMap'
import { mount, shallow } from 'enzyme'
import React from 'react'
import { ZoomWidgetComponent } from '../src/components/document-widgets/ZoomMode'
import { defaultLocalization } from '../src/store/Localization'

function getComponentWithProps(props?: Partial<ZoomWidgetComponent['props']>) {
  return shallow(
    <ZoomWidgetComponent
      localization={defaultLocalization}
      customZoomLevel={0}
      setZoomLevel={() => ({} as any)}
      setZoomMode={() => ({} as any)}
      zoomMode="custom"
      {...props}
    />,
  )
}
describe('ZoomMode component', () => {
  it('should render without crashing', () => {
    const customWrapper1 = getComponentWithProps({ customZoomLevel: 0, zoomMode: 'custom' })
    expect(customWrapper1).toMatchSnapshot()

    const customWrapper2 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'custom' })
    expect(customWrapper2).toMatchSnapshot()

    const customWrapper3 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'fit' })
    expect(customWrapper3).toMatchSnapshot()

    const customWrapper4 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'fitHeight' })
    expect(customWrapper4).toMatchSnapshot()

    const customWrapper5 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'fitWidth' })
    expect(customWrapper5).toMatchSnapshot()

    const customWrapper6 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'originalSize' })
    expect(customWrapper6).toMatchSnapshot()

    const customWrapper7 = getComponentWithProps({ customZoomLevel: 1, zoomMode: 'someOtherValue' } as any)
    expect(customWrapper7).toMatchSnapshot()
  })

  it('should open zoomMenu on icon click', () => {
    const wrapper = getComponentWithProps()
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    expect(wrapper.find(Menu).prop('open')).toBeTruthy()
  })

  it('should close ZoomMenu and sets the zoomMode to fit when ZoomOutMap clicked', () => {
    const setZoomMode = jest.fn()
    const wrapper = getComponentWithProps({ setZoomMode })
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper
      .find(ZoomOutMap)
      .parent()
      .simulate('click')
    expect(setZoomMode).toBeCalledWith('fit')
    expect(wrapper.find(Menu).prop('open')).toBeFalsy()
  })

  it('should close ZoomMenu and sets the zoomMode to originalSize when AspectRatio clicked', () => {
    const setZoomMode = jest.fn()
    const wrapper = getComponentWithProps({ setZoomMode })
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper
      .find(AspectRatio)
      .parent()
      .simulate('click')
    expect(setZoomMode).toBeCalledWith('originalSize')
    expect(wrapper.find(Menu).prop('open')).toBeFalsy()
  })

  it('should close ZoomMenu and sets the zoomMode to fitHeight when reversed "Fit Height" clicked', () => {
    const setZoomMode = jest.fn()
    const wrapper = getComponentWithProps({ setZoomMode })
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper
      .find(Code)
      .first()
      .parent()
      .simulate('click')
    expect(setZoomMode).toBeCalledWith('fitHeight')
    expect(wrapper.find(Menu).prop('open')).toBeFalsy()
  })

  it('should close ZoomMenu and sets the zoomMode to fitWidth when "Fit Width" clicked', () => {
    const setZoomMode = jest.fn()
    const wrapper = getComponentWithProps({ setZoomMode })
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper
      .find(Code)
      .last()
      .parent()
      .simulate('click')
    expect(setZoomMode).toBeCalledWith('fitWidth')
    expect(wrapper.find(Menu).prop('open')).toBeFalsy()
  })

  it('should call setZoomLevel when ZoomIn clicked', async () => {
    const setZoomLevel = jest.fn()
    const wrapper = mount(
      <ZoomWidgetComponent
        localization={defaultLocalization}
        customZoomLevel={0}
        setZoomLevel={setZoomLevel}
        setZoomMode={jest.fn()}
        zoomMode="custom"
      />,
    )
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper.find(ZoomIn).simulate('click')
    expect(setZoomLevel).toBeCalledWith(1)
  })

  it('should call setZoomLevel when ZoomOut clicked', async () => {
    const setZoomLevel = jest.fn()
    const wrapper = mount(
      <ZoomWidgetComponent
        localization={defaultLocalization}
        customZoomLevel={5}
        setZoomLevel={setZoomLevel}
        setZoomMode={jest.fn()}
        zoomMode="custom"
      />,
    )
    wrapper.find(IconButton).simulate('click', { currentTarget: IconButton })
    wrapper.find(ZoomOut).simulate('click')
    expect(setZoomLevel).toBeCalledWith(4)
  })
})
