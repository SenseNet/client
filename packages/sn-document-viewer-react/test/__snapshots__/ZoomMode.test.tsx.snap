// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomMode component should render without crashing 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <ZoomOutIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={0}
      backButton={
        <ForwardRef(WithStyles)
          disabled={true}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 2`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <ZoomInIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 3`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <ZoomOutMapIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 4`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <CodeIcon
      style={
        Object {
          "transform": "rotate(90deg)",
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 5`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <CodeIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 6`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <AspectRatioIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`ZoomMode component should render without crashing 7`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
    title="Zoom mode"
  >
    <ErrorIcon />
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Menu))
    id="zoom-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ZoomOutMapIcon />
         
      Fit
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <AspectRatioIcon />
        
      Original size
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon
        style={
          Object {
            "transform": "rotate(90deg)",
          }
        }
      />
         
      Fit height
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <CodeIcon />
         
      Fit width
       
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(Divider))
      light={true}
    />
      
    Custom
     
    <br />
    <WithStyles(ForwardRef(MobileStepper))
      activeStep={1}
      backButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomOutIcon) />
        </ForwardRef(WithStyles)>
      }
      nextButton={
        <ForwardRef(WithStyles)
          disabled={false}
          onClickCapture={[Function]}
        >
          <Memo(ZoomInIcon) />
        </ForwardRef(WithStyles)>
      }
      position="static"
      steps={6}
      variant="progress"
    />
  </WithStyles(ForwardRef(Menu))>
</div>
`;
